lib/internal/child_process.js

```diff
@@ -912,7 +928,14 @@ function setupChannel(target, channel, serializationMode) {
   };
 
   function emit(event, message, handle) {
-    target.emit(event, message, handle);
+    const isInternalMessage = 'internalMessage' === event;
+    const hasListenersInstalled = target.listenerCount('message');
+    if (hasListenersInstalled || isInternalMessage) {
```

Comment:
We should probably refactor this so it doesn't have to call `listenerCount` when it's not necessary.
```suggestion
    if ('internalMessage' === event || target.listenerCount('message') !== 0) {
```

Reply:
fixed!